### PR TITLE
fix(DCOS-14146): Fix sorting for health status in a couple of places.

### PR DESCRIPTION
**Pull request by weblancaster**
This PR fixes a variety of health sorting issues included in the Jira ticket (DCOS-14146), where now those places the health sorting weight is based on visibility importance, meaning `Unhealthy, NA, Idle/Warn, Healthy`.

Be aware that some places at first loading the sort is selected to be in another field e.g `name`, that being said you will need to select the health field.

- Dashboard pane
- Node Components tab
- Components page
- Node components
- Services (task detail)

To accomplish the fix a couple of refactoring work was needed to be done around how the higher order functions handle the sorting, In summary I have split/removed the complexity in to smaller functions easier to understand and enabling more precise unit tests.

The work in this PR also refactors the work in #2001 which was/is related.

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by MatApple** [comment]
Hey @weblancaster - I've added [WIP] to the beginning of the PR title :) That's how we usually indicate to others that this is WIP 👍

**Comment by weblancaster** [comment]
Ok cool @MatApple thanks.

**Comment by weblancaster** [comment]
retest this please

**Comment by jfurrow** [comment]
I think we need to implement a tie breaker for the dashboard's Component Health panel, as I'm seeing the order change arbitrarily. Check out the following gif:

**Comment by weblancaster** [comment]
@jfurrow you are right. I have pushed a new commit fixing that.

screen shot 2017-04-04 at 9 27 15 am

**Comment by jfurrow** [comment]
Nice @weblancaster! I just noticed the same thing is happening in the Components table also (at the `/components` route), so we'll need the tie breaker there too.

**Comment by weblancaster** [comment]
Ops I thought the id was sufficient for the tie breaker. @jfurrow PR updated and good catches.

screen shot 2017-04-04 at 1 17 19 pm

**Comment by weblancaster** [comment]
retest this please

**Comment by jfurrow** [comment]
Hmm, now I'm getting the following error from the plugins :(

**Comment by weblancaster** [comment]
Looks like @jfurrow screenshot isn't related to the PR. We are investigating.

**Comment by jfurrow** [comment]
Great work @weblancaster!

File: src/js/constants/UnitHealthStatus.js [hunk 1]
```diff
@@ -1,23 +1,52 @@
...
+    value: SERVER_WARN
+  },
+  [SERVER_NA]: {
+    title: 'N/A',
+    key: 'NA',
+    classNames: 'text-mute',
+    sortingValue: 1,
+    value: SERVER_NA
   },
   NA: {
```
Comment (mlunoe) [inline src/js/constants/UnitHealthStatus.js]: Shouldn't this also be `[SERVER_NA]`?

File: src/js/utils/TableUtil.js [hunk 2]
```diff
@@ -71,6 +73,73 @@ var TableUtil = {
...
+   * @returns {Number} HealthSorting value
+   */
+  getHealthSortingValue(healthValue) {
+    let match;
+
+    if (typeof healthValue == 'number') {
+      match = UnitHealthStatus[healthValue].sortingValue;
+    }
+
+    if ( typeof healthValue == 'string' ) {
```
Comment (mlunoe) [inline src/js/utils/TableUtil.js]: #nit let remove the extra spaces inside the `if`-statement
```
if ( typeof healthValue == 'string' ) {
    ^                              ^
```

**Comment by mlunoe** [comment]
This looks and works great! Great job @weblancaster! Thanks for making this change!
Two small comments, I'll wait with merging until they are done

File: src/js/utils/TableUtil.js (hunk 2 above)
Comment (weblancaster) [inline src/js/utils/TableUtil.js]: This is addressed

File: src/js/constants/UnitHealthStatus.js (hunk 1 above)
Comment (weblancaster) [inline src/js/constants/UnitHealthStatus.js]: This is addressed

**Comment by mlunoe** [comment]
Nice work! ⛵️

**Comment by weblancaster** [comment]
thanks for the reviews @mlunoe @jfurrow

**Pull request merged**